src/components/Search.js

```diff
@@ -1,34 +1,116 @@
 import React from 'react';
+import {withRouter} from 'react-router';
+
+import {bindActionCreators} from "redux";
+import {fetchData} from "../redux/actions";
+import {connect} from "react-redux";
 
 import Button from './Button';
 
-export default class Search extends React.PureComponent {
+import {
+    SEARCH_PARAM_NAME,
+    SEARCHBY_PARAM_NAME,
+    SEARCH_TITLE,
+    SEARCH_GENRE
+} from '../constants';
+
+
+class Search extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        const searchParams = new URLSearchParams(this.props.location.search);
+
+        this.state = {
+            searchParams,
+            searchString: searchParams.get(SEARCH_PARAM_NAME) || '',
+            searchBy: searchParams.get(SEARCHBY_PARAM_NAME) || SEARCH_TITLE
+        };
+    }
+
+    handleChange = (event) => {
+        this.setState({
+            ...this.state,
```

Comment:
setState function merges what you pass to it, into existing state object, so there is no need to manually destruct ...this.state here. setState will do it under the hood

Read this article for more details: https://reactjs.org/docs/react-component.html#setstate